tests/unittest/tools/test_test_to_stage_mapping.py

```
@@ -109,6 +110,55 @@ def test_s3_stdout_echo_requires_explicit_opt_in():
+    assert checked, 'Found no documented --stages examples to validate'
+
+
+def test_unknown_stage_reports_a_diagnostic(stage_query):
+    """An unresolvable stage name must not be silently swallowed."""
+    bogus = 'A100X-Triton-Post-Merge-1'
+    assert bogus not in stage_query.stage_to_yaml
+    assert 'A100X-PyTorch-Post-Merge-1' in stage_query.suggest_stages(bogus), \
+        f"Expected the live A100X stage to be suggested for '{bogus}'"
+
+    script = os.path.join(SCRIPTS_DIR, 'test_to_stage_mapping.py')
+    proc = subprocess.run([sys.executable, script, '--stages', bogus],
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)